diff --git a/arch/arm64/boot/dts/freescale/imx8mm-evk.dtsi b/arch/arm64/boot/dts/freescale/imx8mm-evk.dtsi
index 7bb3541a1b41..80a09ce7ec54 100755
--- a/arch/arm64/boot/dts/freescale/imx8mm-evk.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mm-evk.dtsi
@@ -401,6 +401,10 @@ pca6416: gpio@20 {
 ›      ›       reg = <0x20>;
 ›      ›       gpio-controller;
 ›      ›       #gpio-cells = <2>;
+›      ›       interrupt-parent = <&gpio1>;
+›      ›       interrupts = <12 IRQ_TYPE_LEVEL_LOW>;
+›      ›       pinctrl-names = "default";
+›      ›       pinctrl-0 = <&pinctrl_pca6416_int>;
 ›      ›       vcc-supply = <&buck4_reg>;
 ›      };
•
@@ -704,6 +708,11 @@ &gpu {
 };
•
 &iomuxc {
+›      pinctrl_pca6416_int: pca6416_int_grp {
+›      ›       fsl,pins = <
+›      ›       ›       MX8MM_IOMUXC_GPIO1_IO12_GPIO1_IO12›     ›       0x14f
+›      ›       >;
+›      };
 ›      pinctrl_ir_recv: ir-recv {
 ›      ›       fsl,pins = <
 ›      ›       ›       MX8MM_IOMUXC_GPIO1_IO13_GPIO1_IO13›     ›       0x4f
